api/filters.go

```diff
@@ -420,6 +420,72 @@ func (api *API) addFilterDimension(w http.ResponseWriter, r *http.Request) {
 	api.respond.JSON(ctx, w, http.StatusCreated, resp)
 }
 
+func (api *API) updateFilterDimension(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	filterID := chi.URLParam(r, "id")
+	dimensionName := chi.URLParam(r, "name")
+
+	logData := log.Data{
+		"filter_id":      filterID,
+		"dimension_name": dimensionName,
+	}
+
+	updatedDimension := model.Dimension{
+		Options: []string{},
+	}
+
+	if err := api.ParseRequest(r.Body, &updatedDimension); err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusBadRequest,
+			Error{
+				err:     errors.Wrap(err, "failed to parse update filter request"),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	// The new dimension won't be present on the dataset (i.e. only `City` will be present, not `Country`),
+	// so instead of validating it against the existing Version, we check to see if the dimension exists in Cantabular.
+	if err := api.validateCantabularDimensionExists(ctx, filterID, updatedDimension.Name); err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			statusCode(err),
+			Error{
+				err:     errors.Wrapf(err, "error searching for dimension %s in cantabular", updatedDimension.Name),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	var eTag string
+	if reqETag := api.getETag(r); reqETag != eTagAny {
+		eTag = reqETag
+	}
+
+	newETag, err := api.store.UpdateFilterDimension(ctx, filterID, dimensionName, updatedDimension, eTag)
+	if err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			statusCode(err),
+			Error{
+				err:     fmt.Errorf("unable to update filter dimension: %w", err),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	w.Header().Set("ETag", newETag)
```

Comment:
probably best to use the const eTagHeader here.. Actually not sure if it's been added yet, might be in an upcoming PR

Reply:
Doesn't appear to be in yet. Can do a search and replace once the change goes in — think there are a couple of places just using a string.